torchgeo/datasets/inria.py

```diff
@@ -0,0 +1,243 @@
+
+    * Imagery - Aerial GeoTIFFs
+    * Labels - Aerial GeoTIFFs
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://doi.org/10.1109/IGARSS.2017.8127684
+
+
+    .. versionadded:: 0.3
+    """
+
+    foldername = "AerialImageDataset"
+    archive_name = "NEW2-AerialImageDataset.zip"
+    md5 = "4b1acfe84ae9961edc1a6049f940380f"
+
+    def __init__(
+        self,
+        root: str,
+        split: str = "train",
+        transforms: Optional[Callable[[Dict[str, Any]], Dict[str, Any]]] = None,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new InriaBuildings Dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            split: train/test split
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version.
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            RuntimeError: if dataset is missing
+        """
+        self.root = root
+        assert split in {"train", "test"}
+        self.split = split
+        self.transforms = transforms
+        self.checksum = checksum
+
+        self._check_integrity()
+        self.files = self._load_files(root)
+
+    def _load_files(self, root: str) -> List[Dict[str, str]]:
+        """Return the paths of the files in the dataset.
+
+        Args:
+            root: root dir of dataset
+
+        Returns:
+            list of dicts containing paths for each pair of image and label
+        """
+        files = []
+        root_dir = os.path.join(root, self.foldername, self.split)
+        images = glob.glob(os.path.join(root_dir, "images", "*.tif"))
+        images = sorted(images)
+        if self.split == "train":
+            labels = glob.glob(os.path.join(root_dir, "gt", "*.tif"))
+            labels = sorted(labels)
+
+            for img, lbl in zip(images, labels):
+                files.append({"image_path": img, "label_path": lbl})
+        else:
+            for img in images:
+                files.append({"image_path": img})
+
+        return files
+
+    def _load_image(self, path: str) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            the image
+        """
+        filename = os.path.join(path)
+        with rio.open(filename) as img:
+            array = img.read().astype(np.int32)
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            return tensor
+
+    def _load_label(self, path: str) -> Tensor:
+        """Loads a single label.
+
+        Args:
+            path: path to the label
+
+        Returns:
+            the label
+        """
+        filename = os.path.join(path)
+        with rio.open(filename) as img:
+            array = img.read().astype(np.int32)
+            array = array // 255
```

Comment:
nit: Should probably just use `np.clip(array, a_min=0, a_max=1)` here instead since division can be expensive.

Reply:
@isaaccorley these are **very** different operations. If the array contains values ranging from 0 to 255, `np.clip` will convert all values greater than 1 to 1, so the entire image would just be white.